### PR TITLE
(feat) add --version to cli

### LINKED ISSUE #3909: [Feature]: How to programmatically get the version of OpenHands
**What problem or use case are you trying to solve?**

I would like to see the version of OpenHands from another program using the pypi package `openhands-ai`. I have tried:
```python
version = openhands.__version__
```

and
```python
version = pkg_resources.get_distribution('openhands-ai').version
```

and neither seem to work. It'd be nice if there's a way to make this possible!

### PR DESCRIPTION
**Pull request by tobitege**
**Short description of the problem this fixes or functionality that this introduces. This may be used for the CHANGELOG**

Add --version arg to CLI.

---
**Give a summary of what the PR does, explaining any non-trivial design decisions**

Fix #3909 

- Sidenote: `pkg_resources` is deprecated

---

**USAGE**

- In Python:
```
from openhands import __version__
print(__version__)
```

- Using python via poetry (2 variants):
```
$ poetry run python -c "from openhands import __version__; print(__version__)"
0.9.3

$ poetry run python -c "from importlib.metadata import metadata; print(metadata('openhands-ai')['Version'])"
0.9.3
```

- Using CLI:
```
$ poetry run python openhands/core/cli.py --v
INFO:openhands:DEBUG mode enabled.
11:19:29 - openhands:INFO: logger.py:185 - Logging to file in: /mnt/d/github/OpenDevin/logs
0.9.3
```

TODO:
- [ ] How is this supposed to work with OH as an installed package?

### CODE REVIEW

File: openhands/__init__.py [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+    return 'unknown'
+
+
+try:
+    __version__ = get_version()
+except Exception:
+    __version__ = 'unknown'
+
+# ... rest of your __init__.py file ...
```
Comment (enyst) [inline openhands/__init__.py:30]: 😂

Comment (tobitege) [reply openhands/__init__.py:30]: Like Windows: "Unknown Error" :P

File: tests/unit/test_arg_parser.py [hunk 2]
```diff
@@ -129,4 +129,4 @@ def test_help_message(capsys):
         assert element in help_output, f"Expected '{element}' to be in the help message"
 
     option_count = help_output.count('  -')
-    assert option_count == 14, f'Expected 14 options, found {option_count}'
+    assert option_count >= 14, f'Expected 14 options, found {option_count}'
```
[the remark this answers is not in the document]
Comment (tobitege) [reply tests/unit/test_arg_parser.py]: Good catch, artifact from testing. Thanks!

**Review by rbren — APPROVED** [review]
Nice!

**Review by Yoggie9477 — APPROVED** [review]

**Review by Yoggie9477 — APPROVED** [review]

**Review by Yoggie9477 — APPROVED** [review]

**Pull request merged**